map/src/test/java/openlr/map/simplemockdb/SimpleMockedLine.java

```diff
@@ -0,0 +1,231 @@
+            shapePoints[index + 1] = toCartesian(shapePoint.getLongitude(), shapePoint.getLatitude());
+        }
+        shapePoints[numberOfShapePoints - 1] = toCartesian(this.endNode.getLongitudeDeg(), this.endNode.getLatitudeDeg());
+        for (int index = 1; index < shapePoints.length; ++index) {
+            Coordinate end = shapePoints[index];
+            Coordinate start = shapePoints[index - 1];
+            LineSegment segment = new LineSegment(start, end);
+            lineSegments.add(segment);
+        }
+
+    }
+
+
+    public static SimpleMockedLine from(generated.Line xmlLine, Node startNode, Node endNode) {
+        SimpleMockedLine simpleMockedLine = new SimpleMockedLine(xmlLine, startNode, endNode);
+        simpleMockedLine.generateShape();
+        return simpleMockedLine;
+    }
+
+
+    private SimpleMockedLine(generated.Line xmlLine, Node startNode, Node endNode) {
+        this.xmlLine = xmlLine;
+        this.startNode = startNode;
+        this.endNode = endNode;
+        this.lineSegments = new ArrayList<>();
+    }
+
+    public Node getStartNode() {
+        return this.startNode;
+    }
+
+    public List<Long> getRestrictions() {
+        return xmlLine.getRestrictions().stream().map(line -> line.longValue()).collect(Collectors.toList());
+    }
+
+    public Node getEndNode() {
+        return this.endNode;
+    }
+
+    public FormOfWay getFOW() {
+
+        return FormOfWay.getFOWs().get(xmlLine.getFow());
+    }
+
+    public FunctionalRoadClass getFRC() {
+        return FunctionalRoadClass.getFRCs().get(xmlLine.getFrc());
+    }
+
+    /**
+     * @deprecated
+     */
+    @Deprecated
+    public Point2D.Double getPointAlongLine(int var1) {
+        return null;
+    }
+
+    public GeoCoordinates getGeoCoordinateAlongLine(int distance) throws SimpleMockedException {
+
+        int lengthCovered = 0;
+        for (LineSegment segment : lineSegments) {
+            if (lengthCovered <= distance && lengthCovered + segment.getLength() >= distance) {
+                double offset = distance - lengthCovered;
+                Coordinate point = segment.pointAlong(segment.getLength() / offset);
+                point = segment.project(point);
+                double z = segment.p0.z + 1 * (segment.p1.z - segment.p0.z);
+                point.setOrdinate(2, z);
+                return ToWGS84(point);
+            }
+            lengthCovered += segment.getLength();
+        }
+        throw new SimpleMockedException("length is greater than the line length");
+    }
+
+    public int getLineLength() {
+        return (int) this.lineSegments.stream().mapToDouble(segment -> segment.getLength()).sum();
+    }
+
+    public long getID() {
+        return xmlLine.getId().longValue();
+    }
+
+    public Iterator<Line> getPrevLines() {
+        return this.startNode.getIncomingLines();
+    }
+
+    /**
+     * @deprecated
+     */
+    @Deprecated
```

Comment:
Deprecation should indicate the alternative method to get the Shape, maybe in a different format, or indicate that it's not supported any more. You can also throw `java.lang.UnsupportedOperationException` instead of returning `null`.

Reply:
1d4d6793adde69ff0e421f4a2d511bfd44b45c97, 0000374148991942dbe257150ae7e79d7bc32feb